src/Vehicle/Vehicle.cc

```diff
@@ -384,6 +388,9 @@ void Vehicle::_commonInit(void)
 
     _flightDistanceFact.setRawValue(0);
     _flightTimeFact.setRawValue(0);
+
+    _clockUpdateTimer.start(_clockUpdateRateMSecs);
```

Comment:
Instead of adding all of this to vehicle, can you override FactGroup::_updateAllValues in VehicleClockFactGroup and add the QDateTime calls there?